### PR TITLE
[chore] Define code ownership in slightly more open terms

### PR DESCRIPTION
**Pull request by djaglowski**
See #31115 

Our existing definition and process for code ownership are quite good but this PR suggests a few minor tweaks.
- Add triaging as a code owner responsibility. Since this is arguably the easiest thing a code owner can do, we should highlight that it is helpful work.
- Move "The existing Code Owners are actively looking for more help" to the top of the list of reasons why additional code owners may be needed, since this is most often the case.
- Add a note about code ownership not needing to be a full time job. The goal is not to lower the bar in terms of quality, but to encourage lower bandwidth contributors not to self-select out.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (crobert-1) [inline CONTRIBUTING.md:373]:
```suggestion
Code Ownership does not have to be a full time job. If you can find a couple hours to help out on a recurring basis, please consider pursuing Code Ownership.
```

Comment (crobert-1) [inline CONTRIBUTING.md:373]:
```suggestion
Code Ownership does not have to be a full-time job. If you can find a couple hours to help out on a recurring basis, please consider pursing Code Ownership.
```
Tiniest of nits, feel free to ignore.

File: CONTRIBUTING.md
Comment (crobert-1) [inline CONTRIBUTING.md:368]: Is there a way for existing code owners to communicate this? You mentioned in the sig possibly labelling components (IIRC)? Should there be some value that can be specified in `mdatagen.yaml` that could then be added to the component's README to communicate this?

Sorry, I see you're just moving this comment so this isn't meant to block your change, more of a question/discussion.

**Review by andrzej-stencel — APPROVED** [review]

File: CONTRIBUTING.md
Comment (djaglowski) [reply CONTRIBUTING.md:368]: That's the idea. I proposed something detailed in #31115

File: CONTRIBUTING.md
Comment (djaglowski) [inline CONTRIBUTING.md:373]: Consolidating these suggestions into one so I can commit here
```suggestion
Code Ownership does not have to be a full-time job. If you can find a couple hours to help out on a recurring basis, please consider pursuing Code Ownership.
```

**Review by crobert-1 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by djaglowski** [comment]
It would be great to have more feedback or approvals from @open-telemetry/collector-contrib-approvers

**Review by mx-psi — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**
